libraries/Microsoft.Bot.Builder.LanguageGeneration/LGParser.cs

```diff
@@ -1,34 +1,127 @@
+    /// <param name="resourceId">Resource id to resolve.</param>
+    /// <returns>Resolved resource content and unique id.</returns>
+    public delegate (string content, string id) ImportResolverDelegate(string sourceId, string resourceId);
+
+    /// <summary>
+    /// Parser to turn lg content into an <see cref="LGFile"/>.
+    /// </summary>
+    public static class LGParser
     {
-        public static LGResource Parse(string content, string id = "")
+        /// <summary>
+        /// Parser to turn lg content into an <see cref="LGFile"/>.
+        /// </summary>
+        /// <param name="filePath">LG absolute file path.</param>
+        /// <param name="importResolver">resolver to resolve LG import id to template text.</param>
+        /// <returns>new <see cref="LGFile"/> entity.</returns>
+        public static LGFile ParseFile(string filePath, ImportResolverDelegate importResolver = null)
+        {
+            var fullPath = Path.GetFullPath(filePath.NormalizePath());
+            var content = File.ReadAllText(fullPath);
+
+            return ParseContent(content, fullPath, importResolver);
+        }
+
+        /// <summary>
+        /// Parser to turn lg content into an <see cref="LGFile"/>.
+        /// </summary>
+        /// <param name="content">Text content contains lg templates.</param>
+        /// <param name="id">id is the content identifier. If importResolver is null, id must be a full path string. </param>
+        /// <param name="importResolver">resolver to resolve LG import id to template text.</param>
+        /// <returns>new <see cref="LGFile"/> entity.</returns>
+        public static LGFile ParseContent(string content, string id = "", ImportResolverDelegate importResolver = null)
+        {
+            importResolver = importResolver ?? DefaultFileResolver;
+            var lgFile = new LGFile(content: content, id: id, importResolver: importResolver);
+            var diagnostics = new List<Diagnostic>();
+            try
+            {
+                var (templates, imports, errorTemplatesDiagnostics) = AntlrParse(content, id);
+                lgFile.Templates = templates;
+                lgFile.Imports = imports;
+                diagnostics.AddRange(errorTemplatesDiagnostics);
+
+                lgFile.References = GetReferences(lgFile, importResolver);
+                var managedDiagnostics = new StaticChecker(lgFile.AllTemplates.ToList()).Check();
+                diagnostics.AddRange(managedDiagnostics);
+            }
+            catch (LGException ex)
+            {
+                diagnostics.AddRange(ex.Diagnostics);
+            }
+            catch (Exception err)
+            {
+                diagnostics.Add(BuildDiagnostic(err.Message));
+            }
+
+            lgFile.Diagnostics = diagnostics;
+
+            return lgFile;
+        }
+
+        private static (string content, string id) DefaultFileResolver(string filePath, string id)
```

Comment:
can we also name this parameter to (sourceId, resourceId) or (sourceId, dstId), to align with the delegate definition. We can add comments that, this sourceID is sourceFilePath, this dstId, is destinationFilePath). 

Anyhow, the point is, it's unclear to me that what this `id` means, i think we should make it clear.